agatha/construct/semrep_util.py

```diff
@@ -355,6 +383,14 @@ def run(self, input_path:Path, output_path:Path)->None:
     input_path = Path(input_path)
     assert input_path.is_file(), f"Failed to find {input_path}"
     assert not output_path.exists(), f"Refusing to overwrite {output_path}"
+    if line_mode:
+      lines = self._read_lines(input_path)
+      try:
+        out_lines = self._run_gracefully(lines)
+      except ChildProcessError:
+        print("Caught exception, attempting to restart...")
+        # not sure how to make the jump to the next input
```

Comment:
Thought:
```
for line in lines:
  try:
    self._run_gracefully([line])
  except:
     we found the exact error.

Reply:
That is raised in the `_run_gracefully` method right?

Reply:
I mean the `ChildProcessError`

Reply:
Yeah exactly. I should have written `except ChildProcessError` instead. The point I was trying to convey was that you can start to process the input line-by-line if we see an error, and then if you see a _second_ error we will then know the culprit.

Reply:
I'm not sure I follow. How would we trap the `semrep` specific extension since we're simply running the command line tool. When I raise `ChildProcessError` I print the error code and the output of the command on `stderr`. Am I missing something?

Reply:
No, I think this makes sense.
I don't think we really care about the specific error, although it makes sense that we would like to keep it around in the log. What I really care about is what lines of our input file cause errors.